nordic-nrf51822-armcc/CMake/toolchain.cmake

```diff
@@ -0,0 +1,113 @@
+    set(NRF51822_SOFTDEVICE_FILE_PATH    "${CMAKE_CURRENT_LIST_DIR}/../softdevice/s110_nrf51822_8.0.0_softdevice.hex")
+elseif(YOTTA_CFG_NORDIC_SOFTDEVICE STREQUAL "S130")
+    set(NRF51822_LINKER_FLAGS_FILE_PATH  "${CMAKE_CURRENT_LIST_DIR}/../ld/nRF51822_${NRF51822_RAM_SIZE}_S130.sct")
+    set(NRF51822_SOFTDEVICE_FILE_PATH    "${CMAKE_CURRENT_LIST_DIR}/../softdevice/s130_nrf51_1.0.0_softdevice.hex")
+else()
+    message(FATAL_ERROR "SoftDevice version '${YOTTA_CFG_NORDIC_SOFTDEVICE}' is not recognized. Please check your yotta config file.")
+endif()
+
+# append non-generic flags, and set NRF51822-specific link script
+set(_CPU_COMPILATION_OPTIONS "--CPU=Cortex-M0 -D__thumb2__")
+
+set(CMAKE_C_FLAGS_INIT          "${CMAKE_C_FLAGS_INIT} ${_CPU_COMPILATION_OPTIONS}")
+set(CMAKE_ASM_FLAGS_INIT        "${CMAKE_ASM_FLAGS_INIT} ${_CPU_COMPILATION_OPTIONS}")
+set(CMAKE_CXX_FLAGS_INIT        "${CMAKE_CXX_FLAGS_INIT} ${_CPU_COMPILATION_OPTIONS}")
+set(CMAKE_EXE_LINKER_FLAGS_INIT "${CMAKE_EXE_LINKER_FLAGS_INIT} --info=totals --list=.link_totals.txt --scatter ${NRF51822_LINKER_FLAGS_FILE_PATH}")
+
+# used by the apply_target_rules function below:
+set(NRF51822_SOFTDEVICE_HEX_FILE "${NRF51822_SOFTDEVICE_FILE_PATH}")
+set(NRF51822_BOOTLOADER_HEX_FILE "${CMAKE_CURRENT_LIST_DIR}/../bootloader/s130_nrf51_1.0.0_bootloader.hex")
+set(NRF51822_MERGE_HEX_SCRIPT    "${CMAKE_CURRENT_LIST_DIR}/../scripts/merge_hex.py")
+set(NRF51822_MEMORY_INFO_SCRIPT  "${CMAKE_CURRENT_LIST_DIR}/../scripts/memory_info.py")
+
+# define a function for yotta to apply target-specific rules to build products,
+# in our case we need to convert the built elf file to .hex, and add the
+# pre-built softdevice:
+function(yotta_apply_target_rules target_type target_name)
+    if(${target_type} STREQUAL "EXECUTABLE")
+        add_custom_command(OUTPUT ${target_name}.hex ${target_name}-combined.hex
+            DEPENDS ${target_name}
+            # fromelf to hex
+            COMMAND fromelf --i32combined --output=${target_name}.hex ${target_name}
+            # and append the softdevice hex file
+            COMMAND python ${NRF51822_MERGE_HEX_SCRIPT} ${NRF51822_SOFTDEVICE_HEX_FILE} ${target_name}.hex ${target_name}-combined.hex
+            COMMENT "hexifying and adding softdevice to ${target_name}"
+            VERBATIM
+        )
+        if(YOTTA_CFG_IMAGE_FOTA)
+            add_custom_command(OUTPUT ${target_name}-combined-fota.hex
+                DEPENDS ${target_name}.hex
+                # append the softdevice and bootloader hex file
+                COMMAND srec_cat ${NRF51822_SOFTDEVICE_HEX_FILE} -intel ${target_name}.hex -intel ${NRF51822_BOOTLOADER_HEX_FILE} -intel -exclude  0x3FC00 0x3FC20 -generate 0x3FC00 0x3FC04 -constant_little_endian 0x01 4 -generate 0x3FC04 0x3FC08 -constant_little_endian 0x00 4 -generate 0x3FC08 0x3FC0C -constant_little_endian 0xFE 4 -generate 0x3FC0C 0x3FC20 -constant 0x00 -o ${target_name}-combined-fota.hex
```

Comment:
srec_cat isn't needed here. We can re-use merge.py. The options for srec_cat are not necessary; those options have already been applied to the bootloader.hex when it was generated.